--- /var/ida/nextcloud-old/core/l10n/fi.js	2021-10-27 10:25:31.054496677 +0000
+++ /var/ida/nextcloud-old/core/l10n/fi.js.original	2021-10-27 10:25:31.055496679 +0000
@@ -1,8 +1,6 @@
 OC.L10N.register(
     "core",
     {
-    "Note to recipient" : "Viesti vastaanottajalle",
-    "Security" : "Turvallisuus",
     "Please select a file." : "Valitse tiedosto.",
     "File is too big" : "Tiedosto on liian suuri",
     "The selected file is not an image." : "Valittu tiedosto ei ole kuva.",
@@ -188,7 +186,7 @@
     "An internal error occurred." : "Tapahtui sisäinen virhe.",
     "Please try again or contact your administrator." : "Yritä uudestaan tai ota yhteys ylläpitäjään.",
     "Username or email" : "Käyttäjätunnus tai sähköpostiosoite",
-    "Username or\temail" : "Käyttäjätunnus tai\tsähköpostiosoite",
+    "Username or\temail" : "Käyttäjätunnus tai\tsähköposti",
     "Log in" : "Kirjaudu sisään",
     "Logging in …" : "Kirjaudutaan sisään...",
     "Wrong username or password." : "Väärä käyttäjätunnus tai salasana.",
