--- /var/ida/nextcloud/core/templates/layout.base.php	2021-11-29 13:44:49.895860548 +0000
+++ /var/ida/nextcloud/core/templates/layout.base.php.original	2021-11-19 13:00:58.029000000 +0000
@@ -5,7 +5,6 @@
 		<title>
 		<?php p($theme->getTitle()); ?>
 		</title>
-	    <script nonce="<?php p(\OC::$server->getContentSecurityPolicyNonceManager()->getNonce()) ?>" src="/core/js/jquery-2.2.4.js"></script>
 		<meta http-equiv="X-UA-Compatible" content="IE=edge">
 		<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">
 		<meta name="theme-color" content="<?php p($theme->getColorPrimary()); ?>">
