--- /var/ida/nextcloud-new/apps/files/js/fileactionsmenu.js	2021-10-27 15:54:06.722752012 +0000
+++ /var/ida/nextcloud-old/apps/files/js/fileactionsmenu.js	2021-10-27 10:25:30.395495085 +0000
@@ -71,17 +71,35 @@
 		render: function() {
 			var self = this;
 			var fileActions = this._context.fileActions;
+			var inRoot = this._context.dir === '/';
 			var actions = fileActions.getActions(
 				fileActions.getCurrentMimeType(),
 				fileActions.getCurrentType(),
 				fileActions.getCurrentPermissions()
 			);
 
-			var defaultAction = fileActions.getCurrentDefaultFileAction();
+			var defaultAction = fileActions.getDefaultFileAction(
+				fileActions.getCurrentMimeType(),
+				fileActions.getCurrentType(),
+				fileActions.getCurrentPermissions()
+			);
 
 			var items = _.filter(actions, function(actionSpec) {
 				return !defaultAction || actionSpec.name !== defaultAction.name;
 			});
+			items = items.filter(function(item) {
+				if (inRoot) {
+					if (
+						item.name === 'Remove'
+						|| item.name === 'MoveCopy'
+						|| item.name === 'Delete'
+						|| item.name === 'Unshare'
+						|| item.name === 'Rename'
+					)
+					return false;
+				}
+				return true;
+			});
 			items = _.map(items, function(item) {
 				if (_.isFunction(item.displayName)) {
 					item = _.extend({}, item);
