--- /var/ida/nextcloud/apps/files/js/dist/templates.js	2021-11-29 13:51:03.196191826 +0000
+++ /var/ida/nextcloud/apps/files/js/dist/templates.js.original	2021-11-29 13:50:31.904996411 +0000
@@ -27835,7 +27835,6 @@
             F.debug("Templates folder not initialized");
             var e = {
                 attach: function(e) {
-                    return;
                     e.addMenuEntry({
                         id: "template-init",
                         displayName: Object(a.translate)("files", "Set up templates folder"),
