--- /var/ida/nextcloud/lib/public/AppFramework/Http/JSONResponse.php	2021-11-12 07:53:09.603298819 +0000
+++ /var/ida/nextcloud/lib/public/AppFramework/Http/JSONResponse.php.original	2021-11-03 10:50:55.019540300 +0000
@@ -70,7 +70,7 @@
 	 * @throws \Exception If data could not get encoded
 	 */
 	public function render() {
-		$response = json_encode($this->data, JSON_HEX_TAG | JSON_PRETTY_PRINT);
+		$response = json_encode($this->data, JSON_HEX_TAG);
 		if ($response === false) {
 			throw new \Exception(sprintf('Could not json_encode due to invalid ' .
 				'non UTF-8 characters in the array: %s', var_export($this->data, true)));
