--- /var/ida/nextcloud-new/lib/private/L10N/Factory.php	2021-10-27 15:54:10.698762465 +0000
+++ /var/ida/nextcloud-old/lib/private/L10N/Factory.php	2021-10-27 10:25:31.275497211 +0000
@@ -4,20 +4,13 @@
  * @copyright 2016 Roeland Jago Douma <roeland@famdouma.nl>
  * @copyright 2016 Lukas Reschke <lukas@statuscode.ch>
  *
- * @author Arthur Schiwon <blizzz@arthur-schiwon.de>
  * @author Bart Visscher <bartv@thisnet.nl>
- * @author Bjoern Schiessle <bjoern@schiessle.org>
- * @author Christoph Wurst <christoph@winzerhof-wurst.at>
- * @author Georg Ehrke <oc.list@georgehrke.com>
- * @author GretaD <gretadoci@gmail.com>
  * @author Joas Schilling <coding@schilljs.com>
- * @author John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>
  * @author Lukas Reschke <lukas@statuscode.ch>
  * @author Morris Jobke <hey@morrisjobke.de>
  * @author Robin Appelman <robin@icewind.nl>
  * @author Robin McCorkell <robin@mccorkell.me.uk>
  * @author Roeland Jago Douma <roeland@famdouma.nl>
- * @author Thomas Citharel <nextcloud@tcit.fr>
  *
  * @license AGPL-3.0
  *
@@ -43,6 +36,7 @@
 use OCP\IUserSession;
 use OCP\L10N\IFactory;
 use OCP\L10N\ILanguageIterator;
+use \Firebase\JWT\JWT;
 
 /**
  * A factory that generates language instances
@@ -61,26 +55,22 @@
 	/**
 	 * @var array Structure: App => string[]
 	 */
-	protected $availableLanguages = [];
+	protected $availableLanguages = ['en', 'fi', 'sv'];
 
 	/**
 	 * @var array
 	 */
-	protected $localeCache = [];
-
-	/**
-	 * @var array
-	 */
-	protected $availableLocales = [];
+	protected $availableLocales = ['en_US', 'fi_FI', 'sv_FI'];
 
 	/**
 	 * @var array Structure: string => callable
 	 */
 	protected $pluralFunctions = [];
 
-	public const COMMON_LANGUAGE_CODES = [
-		'en', 'es', 'fr', 'de', 'de_DE', 'ja', 'ar', 'ru', 'nl', 'it',
-		'pt_BR', 'pt_PT', 'da', 'fi_FI', 'nb_NO', 'sv', 'tr', 'zh_CN', 'ko'
+	const COMMON_LANGUAGE_CODES = [
+		'en', 'fi', 'sv', 'en_US', 'en_UK', 'en_GB', 'fi_FI', 'sv_SV', 'sv_FI'
+		//'en', 'es', 'fr', 'de', 'de_DE', 'ja', 'ar', 'ru', 'nl', 'it',
+		//'pt_BR', 'pt_PT', 'da', 'fi_FI', 'nb_NO', 'sv', 'tr', 'zh_CN', 'ko'
 	];
 
 	/** @var IConfig */
@@ -120,10 +110,9 @@
 	 * @return \OCP\IL10N
 	 */
 	public function get($app, $lang = null, $locale = null) {
-		return new LazyL10N(function () use ($app, $lang, $locale) {
 			$app = \OC_App::cleanAppId($app);
 			if ($lang !== null) {
-				$lang = str_replace(['\0', '/', '\\', '..'], '', $lang);
+			$lang = str_replace(array('\0', '/', '\\', '..'), '', (string) $lang);
 			}
 
 			$forceLang = $this->config->getSystemValue('force_language', false);
@@ -152,7 +141,6 @@
 			}
 
 			return $this->instances[$lang][$app];
-		});
 	}
 
 	/**
@@ -162,57 +150,28 @@
 	 * @return string language If nothing works it returns 'en'
 	 */
 	public function findLanguage($app = null) {
-		$forceLang = $this->config->getSystemValue('force_language', false);
-		if (is_string($forceLang)) {
-			$this->requestLanguage = $forceLang;
-		}
-
-		if ($this->requestLanguage !== '' && $this->languageExists($app, $this->requestLanguage)) {
-			return $this->requestLanguage;
-		}
 
-		/**
-		 * At this point Nextcloud might not yet be installed and thus the lookup
-		 * in the preferences table might fail. For this reason we need to check
-		 * whether the instance has already been installed
-		 *
-		 * @link https://github.com/owncloud/core/issues/21955
-		 */
-		if ($this->config->getSystemValue('installed', false)) {
-			$userId = !is_null($this->userSession->getUser()) ? $this->userSession->getUser()->getUID() :  null;
-			if (!is_null($userId)) {
-				$userLang = $this->config->getUserValue($userId, 'core', 'lang', null);
-			} else {
-				$userLang = null;
-			}
-		} else {
-			$userId = null;
-			$userLang = null;
+        if (array_key_exists('HTTP_HOST', $_SERVER)) {
+	        $hostname = $_SERVER['HTTP_HOST'];
+	        $domain = substr($hostname, strpos($hostname, ".") + 1);
+	        $prefix = preg_replace('/[^a-zA-Z0-9]/', '_', $domain);
+	        $cookie = $prefix . '_fd_sso_session';
+	        if (array_key_exists($cookie, $_COOKIE)) {
+                $key =\OC::$server->getSystemConfig()->getValue('SSO_KEY');
+		        $session = JWT::decode($_COOKIE[$cookie], $key, array('HS256'));
+				if ($session && $session->language) {
+		            return $session->language;
 		}
-
-		if ($userLang) {
-			$this->requestLanguage = $userLang;
-			if ($this->languageExists($app, $userLang)) {
-				return $userLang;
 			}
 		}
 
 		try {
-			// Try to get the language from the Request
 			$lang = $this->getLanguageFromRequest($app);
-			if ($userId !== null && $app === null && !$userLang) {
-				$this->config->setUserValue($userId, 'core', 'lang', $lang);
-			}
+            if ($lang == 'en' || $lang == 'fi' || $lang == 'sv') {
 			return $lang;
-		} catch (LanguageNotFoundException $e) {
-			// Finding language from request failed fall back to default language
-			$defaultLanguage = $this->config->getSystemValue('default_language', false);
-			if ($defaultLanguage !== false && $this->languageExists($app, $defaultLanguage)) {
-				return $defaultLanguage;
-			}
 		}
+		} catch (LanguageNotFoundException $e) { ; }
 
-		// We could not find any language so fall back to english
 		return 'en';
 	}
 
@@ -223,38 +182,15 @@
 	 * @return null|string
 	 */
 	public function findLocale($lang = null) {
-		$forceLocale = $this->config->getSystemValue('force_locale', false);
-		if (is_string($forceLocale) && $this->localeExists($forceLocale)) {
-			return $forceLocale;
-		}
-
-		if ($this->config->getSystemValue('installed', false)) {
-			$userId = null !== $this->userSession->getUser() ? $this->userSession->getUser()->getUID() :  null;
-			$userLocale = null;
-			if (null !== $userId) {
-				$userLocale = $this->config->getUserValue($userId, 'core', 'locale', null);
-			}
-		} else {
-			$userId = null;
-			$userLocale = null;
-		}
 
-		if ($userLocale && $this->localeExists($userLocale)) {
-			return $userLocale;
+		if ($lang == 'fi') {
+			return 'fi_FI';
 		}
 
-		// Default : use system default locale
-		$defaultLocale = $this->config->getSystemValue('default_locale', false);
-		if ($defaultLocale !== false && $this->localeExists($defaultLocale)) {
-			return $defaultLocale;
+		if ($lang == 'sv'){
+			return 'sv_FI';
 		}
 
-		// If no user locale set, use lang as locale
-		if (null !== $lang && $this->localeExists($lang)) {
-			return $lang;
-		}
-
-		// At last, return USA
 		return 'en_US';
 	}
 
@@ -365,29 +301,6 @@
 	}
 
 	/**
-	 * Return the language to use when sending something to a user
-	 *
-	 * @param IUser|null $user
-	 * @return string
-	 * @since 20.0.0
-	 */
-	public function getUserLanguage(IUser $user = null): string {
-		$language = $this->config->getSystemValue('force_language', false);
-		if ($language !== false) {
-			return $language;
-		}
-
-		if ($user instanceof IUser) {
-			$language = $this->config->getUserValue($user->getUID(), 'core', 'lang', null);
-			if ($language !== null) {
-				return $language;
-			}
-		}
-
-		return $this->config->getSystemValue('default_language', 'en');
-	}
-
-	/**
 	 * @param string $locale
 	 * @return bool
 	 */
@@ -396,14 +309,12 @@
 			return true;
 		}
 
-		if ($this->localeCache === []) {
 			$locales = $this->findAvailableLocales();
-			foreach ($locales as $l) {
-				$this->localeCache[$l['code']] = true;
-			}
-		}
+		$userLocale = array_filter($locales, function($value) use ($locale) {
+			return $locale === $value['code'];
+		});
 
-		return isset($this->localeCache[$locale]);
+		return !empty($userLocale);
 	}
 
 	/**
@@ -505,6 +416,7 @@
 
 		if (($this->isSubDirectory($transFile, $this->serverRoot . '/core/l10n/')
 				|| $this->isSubDirectory($transFile, $this->serverRoot . '/lib/l10n/')
+				|| $this->isSubDirectory($transFile, $this->serverRoot . '/settings/l10n/')
 				|| $this->isSubDirectory($transFile, \OC_App::getAppPath($app) . '/l10n/')
 			)
 			&& file_exists($transFile)) {
@@ -531,7 +443,7 @@
 	 * @return string directory
 	 */
 	protected function findL10nDir($app = null) {
-		if (in_array($app, ['core', 'lib'])) {
+		if (in_array($app, ['core', 'lib', 'settings'])) {
 			if (file_exists($this->serverRoot . '/' . $app . '/l10n/')) {
 				return $this->serverRoot . '/' . $app . '/l10n/';
 			}
@@ -562,8 +474,8 @@
 			$plural = preg_replace('#[^n0-9:\(\)\?\|\&=!<>+*/\%-]#', '', $matches[2]);
 
 			$body = str_replace(
-				[ 'plural', 'n', '$n$plurals', ],
-				[ '$plural', '$n', '$nplurals', ],
+				array( 'plural', 'n', '$n$plurals', ),
+				array( '$plural', '$n', '$nplurals', ),
 				'nplurals='. $nplurals . '; plural=' . $plural
 			);
 
@@ -616,16 +528,7 @@
 	public function getLanguages() {
 		$forceLanguage = $this->config->getSystemValue('force_language', false);
 		if ($forceLanguage !== false) {
-			$l = $this->get('lib', $forceLanguage);
-			$potentialName = $l->t('__language_name__');
-
-			return [
-				'commonlanguages' => [[
-					'code' => $forceLanguage,
-					'name' => $potentialName,
-				]],
-				'languages' => [],
-			];
+			return [];
 		}
 
 		$languageCodes = $this->findAvailableLanguages();
@@ -636,22 +539,22 @@
 		foreach ($languageCodes as $lang) {
 			$l = $this->get('lib', $lang);
 			// TRANSLATORS this is the language name for the language switcher in the personal settings and should be the localized version
-			$potentialName = $l->t('__language_name__');
+			$potentialName = (string) $l->t('__language_name__');
 			if ($l->getLanguageCode() === $lang && $potentialName[0] !== '_') {//first check if the language name is in the translation file
-				$ln = [
+				$ln = array(
 					'code' => $lang,
 					'name' => $potentialName
-				];
+				);
 			} elseif ($lang === 'en') {
-				$ln = [
+				$ln = array(
 					'code' => $lang,
 					'name' => 'English (US)'
-				];
+				);
 			} else {//fallback to language code
-				$ln = [
+				$ln = array(
 					'code' => $lang,
 					'name' => $lang
-				];
+				);
 			}
 
 			// put appropriate languages into appropriate arrays, to print them sorted
