--- /var/ida/nextcloud/3rdparty/composer/autoload_psr4.php	2021-11-09 08:36:47.608842427 +0000
+++ /var/ida/nextcloud/3rdparty/composer/autoload_psr4.php.original	2021-11-09 08:36:30.256000000 +0000
@@ -74,7 +74,6 @@
     'GuzzleHttp\\' => array($vendorDir . '/guzzlehttp/guzzle/src'),
     'Giggsey\\Locale\\' => array($vendorDir . '/giggsey/locale/src'),
     'FG\\' => array($vendorDir . '/fgrosse/phpasn1/lib'),
-    'Firebase\\JWT\\' => array($vendorDir . '/firebase/php-jwt/src'),
     'Egulias\\EmailValidator\\' => array($vendorDir . '/egulias/email-validator/src'),
     'Ds\\' => array($vendorDir . '/php-ds/php-ds/src'),
     'Doctrine\\DBAL\\' => array($vendorDir . '/doctrine/dbal/src'),
