--- /var/ida/nextcloud/apps/settings/templates/settings/personal/security/password.php	2021-11-12 11:07:41.214817889 +0000
+++ /var/ida/nextcloud/apps/settings/templates/settings/personal/security/password.php.original	2021-11-12 11:05:39.477000000 +0000
@@ -21,4 +21,41 @@
  *
  */
 
+script('settings', [
+	'settings',
+	'templates',
+	'vue-settings-personal-security',
+]);
+
+if ($_['passwordChangeSupported']) {
+	script('settings', 'security_password');
+}
+
 ?>
\ No newline at end of file
+<?php if ($_['passwordChangeSupported']) { ?>
+<div id="security-password" class="section">
+	<h2 class="inlineblock"><?php p($l->t('Password'));?></h2>
+	<span id="password-error-msg" class="msg success hidden">Saved</span>
+	<div class="personal-settings-setting-box personal-settings-password-box">
+			<form id="passwordform" method="POST">
+				<label for="pass1" class="hidden-visually"><?php p($l->t('Current password')); ?>: </label>
+				<input type="password" id="pass1" name="oldpassword"
+					   placeholder="<?php p($l->t('Current password'));?>"
+					   autocomplete="off" autocapitalize="none" autocorrect="off" />
+
+				<div class="personal-show-container">
+					<label for="pass2" class="hidden-visually"><?php p($l->t('New password'));?>: </label>
+					<input type="password" id="pass2" name="newpassword"
+						   placeholder="<?php p($l->t('New password')); ?>"
+						   data-typetoggle="#personal-show"
+						   autocomplete="off" autocapitalize="none" autocorrect="off" />
+					<input type="checkbox" id="personal-show" class="hidden-visually" name="show" /><label for="personal-show" class="personal-show-label"></label>
+				</div>
+
+				<input id="passwordbutton" type="submit" value="<?php p($l->t('Change password')); ?>" />
+
+			</form>
+	</div>
+	<span class="msg"></span>
+</div>
+<?php } ?>
