--- /var/ida/nextcloud/apps/settings/templates/settings/personal/security/webauthn.php	2021-11-12 11:07:59.107933787 +0000
+++ /var/ida/nextcloud/apps/settings/templates/settings/personal/security/webauthn.php.original	2021-11-12 11:07:19.975000000 +0000
@@ -1,4 +1,5 @@
 <?php
+declare(strict_types=1);
 /**
  * @copyright Copyright (c) 2020, Roeland Jago Douma <roeland@famdouma.nl>
  *
@@ -20,4 +21,11 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  *
  */
+
+script('settings', [
+	'vue-settings-personal-webauthn',
+]);
+
 ?>
+
+<div id="security-webauthn" class="section"></div>
