--- /var/ida/nextcloud-old/apps/files/js/app.js	2021-10-27 10:25:30.394495083 +0000
+++ /var/ida/nextcloud-old/apps/files/js/app.js.original	2021-10-27 10:25:30.394495083 +0000
@@ -51,9 +51,7 @@
 		initialize: function() {
 			this.navigation = new OCA.Files.Navigation($('#app-navigation'));
 			this.$showHiddenFiles = $('input#showhiddenfilesToggle');
-            // In IDA we always show hidden files
-			// var showHidden = $('#showHiddenFiles').val() === "1";
-			var showHidden = true;
+			var showHidden = $('#showHiddenFiles').val() === "1";
 			this.$showHiddenFiles.prop('checked', showHidden);
 
 
@@ -92,26 +90,22 @@
 					scrollTo: urlParams.scrollto,
 					filesClient: OC.Files.getClient(),
 					multiSelectMenu: [
-						/*
 						{
 							name: 'copyMove',
 							displayName:  t('files', 'Move or copy'),
 							iconClass: 'icon-external',
 						},
-						*/
 						{
 							name: 'download',
 							displayName:  t('files', 'Download'),
 							iconClass: 'icon-download',
 						},
 						OCA.Files.FileList.MultiSelectMenuActions.ToggleSelectionModeAction,
-						/*
 						{
 							name: 'delete',
 							displayName: t('files', 'Delete'),
 							iconClass: 'icon-delete',
 						},
-						*/
 					],
 					sorting: {
 						mode: $('#defaultFileSorting').val(),
@@ -226,9 +220,7 @@
 		 * @returns {undefined}
 		 */
 		_onShowHiddenFilesChange: function() {
-            // In IDA hidden files are always shown
-			//var show = this.$showHiddenFiles.is(':checked');
-			var show = true;
+			var show = this.$showHiddenFiles.is(':checked');
 			this._filesConfig.set('showhidden', show);
 			this._debouncedPersistShowHiddenFilesState();
 		},
@@ -239,9 +231,7 @@
 		 * @returns {undefined}
 		 */
 		_persistShowHiddenFilesState: function() {
-            // In IDA hidden files are always shown
-			//var show = this.$showHiddenFiles.is(':checked');
-			var show = true;
+			var show = this._filesConfig.get('showhidden');
 			$.post(OC.generateUrl('/apps/files/api/v1/showhidden'), {
 				show: show
 			});
