--- /var/ida/nextcloud-old/apps/files/js/fileactions.js	2021-10-27 10:25:30.395495085 +0000
+++ /var/ida/nextcloud-old/apps/files/js/fileactions.js.original	2021-10-27 10:25:30.395495085 +0000
@@ -1,11 +1,4 @@
 /*
- * This file is part of the IDA research data storage service
- *
- * @author   CSC - IT Center for Science Ltd., Espoo Finland <servicedesk@csc.fi>
- * @link     https://research.csc.fi/
- */
-
-/*
  * Copyright (c) 2014
  *
  * This file is licensed under the Affero General Public License version 3
@@ -632,24 +625,6 @@
 				permissions: OC.PERMISSION_UPDATE,
 				iconClass: 'icon-rename',
 				actionHandler: function (filename, context) {
-                    if (context.$file.attr('data-mounttype') === 'shared-root') {
-                        OC.Notification.show(t('files', 'Root project folders may not be modified.'), {type: 'error'});
-                        return;
-                    }
-                    var dir = context.dir || context.fileList.getCurrentDirectory();
-                    var pathname = dir + '/' + filename;
-                    var project = OCA.IDA.Util.extractProjectName(pathname);
-                    var scope = OCA.IDA.Util.stripRootFolder(pathname);
-                    try {
-						var message = OCA.IDA.Util.scopeNotOK(project, scope);
-                        if (message !== false) {
-                            OC.Notification.show(t('ida', message), {type: 'error'});
-                            return;
-                        }
-                    } catch (error) {
-                        OC.Notification.show(error, {type: 'error'});
-                        return;
-                    }
 					context.fileList.rename(filename);
 				}
 			});
@@ -668,24 +643,6 @@
 				permissions: $('#isPublic').val() ? OC.PERMISSION_UPDATE : OC.PERMISSION_READ,
 				iconClass: 'icon-external',
 				actionHandler: function (filename, context) {
-                    if (context.$file.attr('data-mounttype') === 'shared-root') {
-                        OC.Notification.show(t('files', 'Root project folders may not be modified.'), {type: 'error'});
-                        return;
-                    }
-                    var dir = context.dir || context.fileList.getCurrentDirectory();
-                    var pathname = dir + '/' + filename;
-                    var project = OCA.IDA.Util.extractProjectName(pathname);
-                    var scope = OCA.IDA.Util.stripRootFolder(pathname);
-                    try {
-						var message = OCA.IDA.Util.scopeNotOK(project, scope);
-                        if (message !== false) {
-                            OC.Notification.show(t('ida', message), {type: 'error'});
-                            return;
-                        }
-                    } catch (error) {
-                        OC.Notification.show(error, {type: 'error'});
-                        return;
-                    }
 					var permissions = context.fileInfoModel.attributes.permissions;
 					var actions = OC.dialogs.FILEPICKER_TYPE_COPY;
 					if (permissions & OC.PERMISSION_UPDATE) {
@@ -697,41 +654,9 @@
 					}
 					OC.dialogs.filepicker(t('files', 'Choose target folder'), function(targetPath, type) {
 						if (type === OC.dialogs.FILEPICKER_TYPE_COPY) {
-                           if (targetPath == null || targetPath == '' || targetPath == '/' || OCA.IDA.Util.testIfFrozen(targetPath)) {
-                                OC.Notification.show(t('ida', 'Files can be added only in the Staging area (root folder ending in +)'), {type: 'error'});
-                                return;
-                            }
-                            var project = OCA.IDA.Util.extractProjectName(targetPath);
-                            var scope = OCA.IDA.Util.stripRootFolder(targetPath);
-                            try {
-						        var message = OCA.IDA.Util.scopeNotOK(project, scope);
-                                if (message !== false) {
-                                    OC.Notification.show(t('ida', message), {type: 'error'});
-                                    return;
-                                }
-                            } catch (error) {
-                                OC.Notification.show(error, {type: 'error'});
-                                return;
-                            }
 							context.fileList.copy(filename, targetPath, false, context.dir);
 						}
 						if (type === OC.dialogs.FILEPICKER_TYPE_MOVE) {
-                           if (targetPath == null || targetPath == '' || targetPath == '/' || OCA.IDA.Util.testIfFrozen(targetPath)) {
-                                OC.Notification.show(t('ida', 'Files can be added only in the Staging area (root folder ending in +)'), {type: 'error'});
-                                return;
-                            }
-                            var project = OCA.IDA.Util.extractProjectName(targetPath);
-                            var scope = OCA.IDA.Util.stripRootFolder(targetPath);
-                            try {
-						        var message = OCA.IDA.Util.scopeNotOK(project, scope);
-                                if (message !== false) {
-                                    OC.Notification.show(t('ida', message), {type: 'error'});
-                                    return;
-                                }
-                            } catch (error) {
-                                OC.Notification.show(error, {type: 'error'});
-                                return;
-                            }
 							context.fileList.move(filename, targetPath, false, context.dir);
 						}
 						context.fileList.dirInfo.dirLastCopiedTo = targetPath; 
@@ -773,33 +698,12 @@
 				// permission is READ because we show a hint instead if there is no permission
 				permissions: OC.PERMISSION_DELETE,
 				iconClass: 'icon-delete',
-				actionHandler: function(filename, context) {
-                    if (context.$file.attr('data-mounttype') === 'shared-root') {
-                        OC.Notification.show(t('files', 'Root project folders may not be modified.'), {type: 'error'});
-                        return;
-                    }
-                    var dir = context.dir || context.fileList.getCurrentDirectory();
-                    var pathname = dir + '/' + filename;
-                    var project = OCA.IDA.Util.extractProjectName(pathname);
-                    var scope = OCA.IDA.Util.stripRootFolder(pathname);
-                    try {
-						var message = OCA.IDA.Util.scopeNotOK(project, scope);
-                        if (message !== false) {
-                            OC.Notification.show(t('ida', message), {type: 'error'});
-                            return;
-                        }
-                    } catch (error) {
-                        OC.Notification.show(error, {type: 'error'});
-                        return;
-                    }
+				actionHandler: function(fileName, context) {
 					// if there is no permission to delete do nothing
 					if((context.$file.data('permissions') & OC.PERMISSION_DELETE) === 0) {
 						return;
 					}
-                    var response = confirm(t('ida', 'Are you sure you want to delete the selected item? THIS ACTION CANNOT BE UNDONE!'));
-                    if (response) {
-					    context.fileList.do_delete(filename, context.dir);
-                    }
+					context.fileList.do_delete(fileName, context.dir);
 					$('.tipsy').remove();
 				}
 			});
