--- /var/ida/nextcloud/apps/files/templates/index.php	2021-11-29 13:44:49.846860242 +0000
+++ /var/ida/nextcloud/apps/files/templates/index.php.original	2021-11-12 08:41:14.103000000 +0000
@@ -2,8 +2,6 @@
 <?php $_['appNavigation']->printPage(); ?>
 <div id="app-content">
 
-    <meta name="fdwe-scope" content="FILES">
-
 	<?php if (!$_['isIE']) { ?>
 		<input type="checkbox" class="hidden-visually" id="showgridview"
 			aria-label="<?php p($l->t('Toggle grid view'))?>"
