--- /var/ida/nextcloud/apps/files/templates/list.php	2021-11-11 14:13:58.600745311 +0000
+++ /var/ida/nextcloud/apps/files/templates/list.php.original	2021-11-03 10:50:54.911539647 +0000
@@ -5,7 +5,8 @@
 		</div>
 		<div id="file_action_panel"></div>
 		<div class="notCreatable notPublic hidden">
-            <?php p($l->t('Files can be added only in the Staging area (root folder ending in +)'))?>
+			<div class="icon-alert-outline"></div>
+			<?php p($l->t('You don’t have permission to upload or create files here'))?>
 		</div>
 	<?php /* Note: the template attributes are here only for the public page. These are normally loaded
 			 through ajax instead (updateStorageStatistics).
