--- /var/ida/nextcloud/apps/files_sharing/lib/Controller/ShareAPIController.php	2021-11-29 13:44:49.866860367 +0000
+++ /var/ida/nextcloud/apps/files_sharing/lib/Controller/ShareAPIController.php.original	2021-11-03 10:50:54.898539568 +0000
@@ -400,13 +400,6 @@
 			throw new OCSNotFoundException($this->l->t('Wrong share ID, share doesn\'t exist'));
 		}
 
-		if (($share->getShareType() === Share::SHARE_TYPE_GROUP ||
-		    $share->getShareType() === Share::SHARE_TYPE_ROOM) &&
-	        $share->getShareOwner() !== $this->currentUser &&
-	        $share->getSharedBy() !== $this->currentUser) {
-	        // If node is a shared project folder, do nothing and throw exception
-	        throw new OCSForbiddenException($this->l->t('Root project folders may not be modified.'));
-        } else {
 		    // if it's a group share or a room share
 		    // we don't delete the share, but only the
 		    // mount point. Allowing it to be restored
@@ -417,7 +410,6 @@
 			    if (!$this->canDeleteShare($share)) {
 				    throw new OCSForbiddenException($this->l->t('Could not delete share'));
 			    }
-			}
 
 			$this->shareManager->deleteShare($share);
 		}
