--- /var/ida/nextcloud/apps/files/css/files.scss	2021-11-11 13:09:57.560924038 +0000
+++ /var/ida/nextcloud/apps/files/css/files.scss.original	2021-11-03 10:50:54.804538998 +0000
@@ -467,7 +467,7 @@
 
 /* File checkboxes */
 #fileList tr td.selection>.selectCheckBox + label:before {
-	opacity: 1;
+	opacity: 0.3;
 	margin-right: 0;
 }
 
@@ -609,7 +609,7 @@
 		display: inline;
 		padding: 17px 8px;
 		line-height: 50px;
-		opacity: 1;
+		opacity: .3;
 		&.action-share {
 			padding: 17px 14px;
 			&.permanent:not(.shared-style) .icon-shared + span {
@@ -635,18 +635,18 @@
 		}
 		&.no-permission {
 			&:hover, &:focus {
-				opacity: 1;
+				opacity: .3;
 			}
 		}
 		&.disabled {
 			&:hover, &:focus,
 			img {
-				opacity: 1;
+				opacity: .3;
 			}
 			&.action-download {
-				opacity: 1;
+				opacity: .7;
 				&:hover, &:focus {
-					opacity: 1;
+					opacity: .7;
 				}
 			}
 		}
@@ -659,7 +659,7 @@
 		}
 	}
 	.fileActionsMenu a.action, a.action.action-share.shared-style {
-		opacity: 1;
+		opacity: .7;
 	}
 	.fileActionsMenu .action.permanent {
 		opacity: 1;
@@ -707,7 +707,6 @@
 
 .summary {
 	color: var(--color-text-maxcontrast);
-	opacity: 1;
 	/* add whitespace to bottom of files list to correctly show dropdowns */
 	$action-menu-items-count: 7; // list view has currently max 7 items in its action menu 
 	height: 44px * ($action-menu-items-count + 0.5); // 0.5 is added to show some whitespace below
