--- /var/ida/nextcloud-new/lib/public/AppFramework/Http/JSONResponse.php	2021-10-27 15:54:10.723762531 +0000
+++ /var/ida/nextcloud-old/lib/public/AppFramework/Http/JSONResponse.php.original	2021-10-27 10:25:31.302497276 +0000
@@ -3,10 +3,8 @@
  * @copyright Copyright (c) 2016, ownCloud, Inc.
  *
  * @author Bernhard Posselt <dev@bernhard-posselt.com>
- * @author Christoph Wurst <christoph@winzerhof-wurst.at>
  * @author Lukas Reschke <lukas@statuscode.ch>
  * @author Morris Jobke <hey@morrisjobke.de>
- * @author Roeland Jago Douma <roeland@famdouma.nl>
  * @author Thomas Müller <thomas.mueller@tmit.eu>
  * @author Thomas Tanghus <thomas@tanghus.net>
  *
@@ -54,9 +52,7 @@
 	 * @param int $statusCode the Http status code, defaults to 200
 	 * @since 6.0.0
 	 */
-	public function __construct($data = [], $statusCode = Http::STATUS_OK) {
-		parent::__construct();
-
+	public function __construct($data=array(), $statusCode=Http::STATUS_OK) {
 		$this->data = $data;
 		$this->setStatus($statusCode);
 		$this->addHeader('Content-Type', 'application/json; charset=utf-8');
