--- /var/ida/nextcloud/core/l10n/fi.json	2021-11-29 13:44:49.893860536 +0000
+++ /var/ida/nextcloud/core/l10n/fi.json.original	2021-11-03 10:50:54.965539973 +0000
@@ -1,19 +1,4 @@
 { "translations": {
-    "Project Suspended": "Project Suspended",
-    "Failed Actions": "Epäonnistuneet toiminnot",
-    "Pending Actions": "Odottavat toiminnot",
-    "This is Fairdata IDA": "Tämä on Fairdata IDA",
-    "Information": "Tietoa",
-    "Accessibility": "Saautettavuus",
-    "Contact": "Ota yhteyttä",
-    "Follow": "Seuraa",
-    "Terms and Policies": "Käyttöpolitiikat ja ehdot",
-    "Contracts and Privacy": "Sopimukset ja tietosuoja",
-    "What's new": "Uutiset",
-    "Accessibility statement": "Saavutettavuus",
-    "Sitemap": "Sivukartta",
-    "Login": "Kirjaudu",
-    "The Fairdata services are offered by the<strong> Ministry of Education and Culture </strong>and produced by<strong> CSC – IT Center for Science Ltd.</strong>": "Fairdata-palvelut järjestää<strong> opetus- ja kulttuuriministerö </strong>ja toimittaa<strong> CSC – Tieteen tietotekniikan keskus Oy</strong>",
     "Please select a file." : "Valitse tiedosto.",
     "File is too big" : "Tiedosto on liian suuri",
     "The selected file is not an image." : "Valittu tiedosto ei ole kuva.",
