--- /var/ida/nextcloud/core/l10n/sv.js	2021-11-29 13:44:49.894860542 +0000
+++ /var/ida/nextcloud/core/l10n/sv.js.original	2021-11-11 17:11:14.817000000 +0000
@@ -1,19 +1,6 @@
 OC.L10N.register(
     "core",
     {
-    "The public link will expire no later than {days} days after it is created, unless a shorter period is specified." : "Den öppna länken föråldras senast {days} dagar efter att den skapats, om inte tiden definieras kortare.",
-    "Login": "Logga in",
-    "Log in": "Logga in",
-    "Username" : "Användarnamn",
-    "Password" : "Lösenord",
-    "Login with CSC account": "Logga in med CSC-användarkonto",
-    "Personal CSC accounts are created and managed in": "Hantering av CSC-profilen sker i",
-    "CSC Customer Portal": "CSC:s kundportal",
-    "CSC username" : "CSC-användarnamn",
-    "CSC password" : "CSC-lösenord",
-    "Enter temporary share password:": "Ange lösenord för tillfällig länk:",
-    "Upload some content!" : "Ladda upp innehåll!",
-    "Files can be added only in the Staging area (root folder ending in +)": "Filer kan läggas till endast i mappar med plustecken (+, område för preparering)",
     "Please select a file." : "Vänligen välj en fil.",
     "File is too big" : "Filen är för stor",
     "The selected file is not an image." : "Den valda filen är ingen bild.",
