--- /var/ida/nextcloud-new/core/css/guest.css	2021-10-27 15:54:09.661759739 +0000
+++ /var/ida/nextcloud-old/core/css/guest.css.original	2021-10-27 10:25:30.758495962 +0000
@@ -17,18 +17,16 @@
 body {
 	background-color: #ffffff;
 	font-weight: normal;
-	/* bring the default font size up to 14px */
-	font-size: .875em;
+	font-size: .8em;
 	line-height: 1.6em;
-	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
+	font-family: 'Nunito', 'Open Sans', Frutiger, Calibri, 'Myriad Pro', Myriad, sans-serif;
 	color: #000;
 	text-align: center;
 	background-color: #0082c9;
 	background-image: url('../img/background.png?v=2');
-	background-image: url('../img/background.png?v=2'), linear-gradient(40deg, #0082c9 0%, rgba(28,175,255,1) 100%);
 	background-position: 50% 50%;
-	background-repeat: repeat;
-	background-size: 275px, contain;
+	background-repeat: no-repeat;
+	background-size: cover;
 	background-attachment: fixed; /* fix background gradient */
 	min-height: 100%; /* fix sticky footer */
 	height: auto;
@@ -97,22 +95,21 @@
 	max-width: 100%;
 	max-height: 200px;
 }
-
 .wrapper {
-	width: 100%;
-	max-width: 700px;
+	width: 300px;
 	margin-top: 10vh;
 }
 
 /* Default FORM */
 form {
 	position: relative;
+	width: 280px;
 	margin: auto;
 	padding: 0;
 }
 form fieldset {
-	width: 260px;
-	margin: auto auto 20px;
+	margin-bottom: 20px;
+	text-align: left;
 	-webkit-user-select: none;
 	-moz-user-select: none;
 	-ms-user-select: none;
@@ -146,8 +143,7 @@
 
 /* Buttons and input */
 #submit-wrapper,
-#reset-password-wrapper,
-#alternative-logins {
+#reset-password-wrapper {
 	display: flex;
 	align-items: center;
 	justify-content: center;
@@ -155,51 +151,7 @@
 						   absolutely positioned descendant icons */
 }
 
-#alternative-logins {
-	margin: auto;
-	display: block;
-	min-width: 260px;
-	max-width: 400px;
-	overflow: hidden;
-}
-
-#alternative-logins a {
-	margin: 10px 5px;
-	display: block;
-	font-size: 15px;
-	white-space: nowrap;
-	overflow: hidden;
-	text-overflow: ellipsis;
-}
-
-#alternative-logins a.button::before {
-	content: "";
-	background-repeat: no-repeat;
-	background-size: contain;
-	width: 0;
-	margin-right: 0;
-	height: 18px;
-	display: inline-block;
-	vertical-align: bottom;
-}
-
-#alternative-logins .button {
-	color: #0082c9;
-	padding: 12px 20px;
-}
-
-@media only screen and (max-width: 1024px) {
-	.wrapper {
-		margin-top: 0;
-	}
-	#alternative-logins {
-		margin: auto;
-	}
-}
-
-
-#submit-wrapper .submit-icon,
-#reset-password-wrapper .submit-icon {
+#submit-wrapper .submit-icon {
 	position: absolute;
 	top: 22px;
 	right: 24px;
@@ -210,10 +162,6 @@
 							 applied to the button instead. */
 }
 
-#submit-wrapper {
-	margin: 0 auto;
-}
-
 #submit-wrapper input.login:hover ~ .submit-icon.icon-confirm-white,
 #submit-wrapper input.login:focus ~ .submit-icon.icon-confirm-white,
 #submit-wrapper input.login:active ~ .submit-icon.icon-confirm-white {
@@ -226,6 +174,10 @@
 	text-overflow: ellipsis;
 }
 
+#reset-password-wrapper .submit-icon {
+	display: none;
+}
+
 #submit-wrapper .icon-loading-small {
 	position: absolute;
 	top: 22px;
@@ -234,7 +186,7 @@
 
 
 input, textarea, select, button, div[contenteditable=true] {
-	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
+	font-family: 'Nunito', 'Open Sans', Frutiger, Calibri, 'Myriad Pro', Myriad, sans-serif;
 }
 input,
 input:not([type='range']),
@@ -279,21 +231,6 @@
 	border: none;
 	font-weight: normal;
 }
-input[type='password'].password-with-toggle, input[type='text'].password-with-toggle {
-	width: 219px;
-	padding-right: 40px;
-}
-.toggle-password {
-	position: absolute;
-	top: 1px;
-	right: 5px;
-	padding: 14px;
-	height: 16px;
-}
-.toggle-password:hover,
-.toggle-password:focus {
-	opacity: .6;
-}
 input.login {
 	width: 260px;
 	height: 50px;
@@ -304,8 +241,6 @@
 input.updateButton,
 input.update-continue {
 	padding: 10px 20px; /* larger log in and installation buttons */
-	overflow: hidden;
-	text-overflow: ellipsis;
 }
 .updateAnyways a.updateAnywaysButton {
 	font-size: 14px;
@@ -340,9 +275,7 @@
 button.primary:not(:disabled):focus,
 a.primary:not(:disabled):hover,
 a.primary:not(:disabled):focus {
-	color: rgba(255, 255, 255, 1);
-	background-image: linear-gradient(40deg, #0082c9 0%, #30b6ff 100%);
-	background-position: initial;
+	color: rgba(255, 255, 255, .8);
 }
 
 /* Checkboxes - white only for login */
@@ -402,6 +335,7 @@
 .strengthify-wrapper {
 	display: inline-block;
 	position: relative;
+	left: 5px;
 	top: -20px;
 	width: 269px;
 	border-radius: 0 0 3px 3px;
@@ -478,10 +412,6 @@
 	color: #fff !important;
 }
 
-.body-login-container.two-factor {
-	max-width: 290px;
-	margin: 15px auto 0;
-}
 .two-factor-provider {
 	display: flex;
 	border-radius: 3px; /* --border-radius */
@@ -537,24 +467,22 @@
 #remember_login {
 	margin: 18px 5px 0 16px !important;
 }
+.remember-login-container,
 .lost-password-container {
 	display: inline-block;
 	margin: 10px 0;
 	text-align: center;
 	width: 100%;
+	text-shadow: 0 0 2px rgba(0, 0, 0, .4); /* better readability on bright background */
 }
-#lost-password,
-#lost-password-back,
-#reset-password-wrapper + a {
+.lost-password-container #lost-password,
+.lost-password-container #lost-password-back {
 	display: inline-block;
-	font-weight: normal !important;
+	font-weight: normal;
 	padding: 12px;
+	margin-top: -6px;
 	color: #fff;
 	cursor: pointer;
-	text-shadow: 0 0 2px rgba(0, 0, 0, .4); /* better readability on bright background */
-}
-#lost-password {
-	margin-top: -6px;
 }
 #forgot-password {
 	padding: 11px;
@@ -656,11 +584,6 @@
 	box-shadow: none !important;
 }
 
-#install-recommended-apps + label span {
-	display: inline-block;
-	opacity: .7;
-}
-
 /* Errors */
 /* Warnings and errors are the same */
 .body-login-container,
@@ -725,6 +648,7 @@
 }
 .error-wide {
 	width: 700px;
+	margin-left: -200px !important;
 	margin-top: 35px;
 }
 .error-wide .button {
@@ -918,7 +842,7 @@
 .hidden-visually {
 	position: absolute;
 	left: -10000px;
-	top: -10000px;
+	top: auto;
 	width: 1px;
 	height: 1px;
 	overflow: hidden;
@@ -927,3 +851,48 @@
 a.legal {
 	font-size: smaller;
 }
+
+
+/* latin-ext */
+@font-face {
+	font-family: 'Nunito';
+	font-style: normal;
+	font-weight: 400; /* normal */
+	src: local('Nunito Regular'), local('Nunito-Regular'),
+		url('../fonts/Nunito-Regular-latin-ext.woff2') format('woff2'),
+		url('../fonts/Nunito-Regular.ttf') format('truetype');
+	unicode-range: U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
+}
+
+/* latin */
+@font-face {
+	font-family: 'Nunito';
+	font-style: normal;
+	font-weight: 400; /* normal */
+	src: local('Nunito Regular'), local('Nunito-Regular'),
+		url('../fonts/Nunito-Regular.woff2') format('woff2'),
+		url('../fonts/Nunito-Regular.ttf') format('truetype');
+	unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
+}
+
+/* latin-ext */
+@font-face {
+	font-family: 'Nunito';
+	font-style: normal;
+	font-weight: 700; /* bold */
+	src: local('Nunito Bold'), local('Nunito-Bold'),
+		url('../fonts/Nunito-Bold-latin-ext.woff2') format('woff2'),
+		url('../fonts/Nunito-Bold.ttf') format('truetype');
+	unicode-range: U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
+}
+
+/* latin */
+@font-face {
+	font-family: 'Nunito';
+	font-style: normal;
+	font-weight: 700; /* bold */
+	src: local('Nunito Bold'), local('Nunito-Bold'),
+		url('../fonts/Nunito-Bold.woff2') format('woff2'),
+		url('../fonts/Nunito-Bold.ttf') format('truetype');
+	unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
+}
