--- /var/ida/nextcloud-new/core/css/variables.scss	2021-10-27 15:54:09.662759741 +0000
+++ /var/ida/nextcloud-old/core/css/variables.scss.original	2021-10-27 10:25:30.760495967 +0000
@@ -35,16 +35,11 @@
 $color-main-background: #fff !default;
 $color-main-background-translucent: rgba($color-main-background, .97) !default;
 
-// used for different active/hover/focus/disabled states
-$color-background-hover: nc-darken($color-main-background, 4%) !default;
+// used for different active/disabled states
 $color-background-dark: nc-darken($color-main-background, 7%) !default;
 $color-background-darker: nc-darken($color-main-background, 14%) !default;
 
-$color-placeholder-light: nc-darken($color-main-background, 10%) !default;
-$color-placeholder-dark: nc-darken($color-main-background, 20%) !default;
-
 $color-primary: #0082c9 !default;
-$color-primary-light: mix($color-primary, $color-main-background, 10%) !default;
 $color-primary-text: #ffffff !default;
 // do not use nc-darken/lighten in case of overriding because
 // primary-text is independent of color-main-text
@@ -63,19 +58,17 @@
 // rgb(118, 118, 118) / #767676
 // min. color contrast for normal text on white background according to WCAG AA
 // (Works as well: color: #000; opacity: 0.57;)
-$color-text-maxcontrast: nc-lighten($color-main-text, 33%) !default;
-$color-text-light: $color-main-text !default;
-$color-text-lighter: $color-text-maxcontrast !default;
+$color-text-maxcontrast: nc-lighten($color-main-text, 46.2%) !default;
+$color-text-light: nc-lighten($color-main-text, 15%) !default;
+$color-text-lighter: nc-lighten($color-main-text, 30%) !default;
 
 $image-logo: url('../img/logo/logo.svg?v=1') !default;
 $image-login-background: url('../img/background.png?v=2') !default;
-$image-logoheader: url('../img/logo/logo.svg?v=1') !default;
-$image-favicon: url('../img/logo/logo.svg?v=1') !default;
 
 $color-loading-light: #ccc !default;
 $color-loading-dark: #444 !default;
 
-$color-box-shadow: transparentize(nc-darken($color-main-background, 70%), 0.5) !default;
+$color-box-shadow: rgba(nc-darken($color-main-background, 70%), 0.5) !default;
 
 // light border like file table or app-content list
 $color-border: nc-darken($color-main-background, 7%) !default;
@@ -86,10 +79,7 @@
 // Pill-style button, value is large so big buttons also have correct roundness
 $border-radius-pill: 100px !default;
 
-$font-face: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Cantarell, Ubuntu, 'Helvetica Neue', Arial, 'Noto Color Emoji', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol' !default;
-$default-font-size: 15px;
-
-$default-line-height: 24px;
+$font-face: 'Nunito', 'Open Sans', Frutiger, Calibri, 'Myriad Pro', Myriad, sans-serif !default;
 
 $animation-quick: 100ms;
 $animation-slow: 300ms;
@@ -102,5 +92,5 @@
 $list-min-width: 200px;
 $list-max-width: 300px;
 
-// mobile. Keep in sync with core/js/js.js
-$breakpoint-mobile: 1024px;
+// mobile
+$breakpoint-mobile: 768px;
