--- /var/ida/nextcloud-old/apps/files_sharing/lib/Controller/ShareAPIController.php	2021-10-27 10:25:30.494495324 +0000
+++ /var/ida/nextcloud-old/apps/files_sharing/lib/Controller/ShareAPIController.php.original	2021-10-27 10:25:30.494495324 +0000
@@ -1,11 +1,4 @@
 <?php
-/*
- * This file is part of the IDA research data storage service
- *
- * @author   CSC - IT Center for Science Ltd., Espoo Finland <servicedesk@csc.fi>
- * @link     https://research.csc.fi/
- */
-
 declare(strict_types=1);
 /**
  * @copyright Copyright (c) 2016, ownCloud, Inc.
@@ -353,8 +346,7 @@
 				$share->getShareType() === Share::SHARE_TYPE_ROOM) &&
 			$share->getShareOwner() !== $this->currentUser &&
 			$share->getSharedBy() !== $this->currentUser) {
-            // If node is a shared project folder, do nothing and throw exception
-            throw new OCSForbiddenException($this->l->t('Root project folders may not be modified.'));
+			$this->shareManager->deleteFromSelf($share, $this->currentUser);
 		} else {
 			$this->shareManager->deleteShare($share);
 		}
