--- /var/ida/nextcloud-old/core/templates/login.php	2021-10-27 10:25:31.189497003 +0000
+++ /var/ida/nextcloud-old/core/templates/login.php.original	2021-10-27 10:25:31.189497003 +0000
@@ -1,13 +1,8 @@
-<?php
-
-/** @var $l \OCP\IL10N */ ?>
+<?php /** @var $l \OCP\IL10N */ ?>
 <?php
 script('core', 'merged-login');
 
 use OC\Core\Controller\LoginController;
-
-$CURRENT_LANGUAGE = $l->getLanguageCode();
-$CURRENT_LANGUAGE = $CURRENT_LANGUAGE ? substr($CURRENT_LANGUAGE, 0, 2) : 'en';
 ?>
 
 <!--[if IE 8]><style>input[type="checkbox"]{padding:0;}</style><![endif]-->
@@ -34,22 +29,33 @@
 			</div>
 		<?php endif; ?>
 		<div id="message" class="hidden">
-			<img class="float-spinner" alt="" src="<?php p(image_path('core', 'loading-dark.gif')); ?>">
+			<img class="float-spinner" alt=""
+				src="<?php p(image_path('core', 'loading-dark.gif'));?>">
 			<span id="messageText"></span>
 			<!-- the following div ensures that the spinner is always inside the #message div -->
 			<div style="clear: both;"></div>
 		</div>
-
-		<p class="login-field<?php if (!empty($_[LoginController::LOGIN_MSG_INVALIDPASSWORD])) { ?> shake<?php } ?>">
-			<input type="text" name="user" id="user" placeholder="Username" value="<?php p($_['loginName']); ?>" <?php p($_['user_autofocus'] ? 'autofocus' : ''); ?> autocomplete="<?php p($_['login_form_autocomplete']); ?>" autocapitalize="none" autocorrect="off" required>
-			<label for="user" class="infield">Username</label>
+		<p class="grouptop<?php if (!empty($_[LoginController::LOGIN_MSG_INVALIDPASSWORD])) { ?> shake<?php } ?>">
+			<input type="text" name="user" id="user"
+				placeholder="<?php p($l->t('Username or email')); ?>"
+				aria-label="<?php p($l->t('Username or email')); ?>"
+				value="<?php p($_['loginName']); ?>"
+				<?php p($_['user_autofocus'] ? 'autofocus' : ''); ?>
+				autocomplete="<?php p($_['login_form_autocomplete']); ?>" autocapitalize="none" autocorrect="off" required>
+			<label for="user" class="infield"><?php p($l->t('Username or email')); ?></label>
 		</p>
-		<p class="login-field<?php if (!empty($_[LoginController::LOGIN_MSG_INVALIDPASSWORD])) { ?> shake<?php } ?>">
-			<input type="password" name="password" id="password" value="" placeholder="Password" <?php p($_['user_autofocus'] ? '' : 'autofocus'); ?> autocomplete="<?php p($_['login_form_autocomplete']); ?>" autocapitalize="none" autocorrect="off" required>
-			<label for="password" class="infield">Password</label>
+
+		<p class="groupbottom<?php if (!empty($_[LoginController::LOGIN_MSG_INVALIDPASSWORD])) { ?> shake<?php } ?>">
+			<input type="password" name="password" id="password" value=""
+				placeholder="<?php p($l->t('Password')); ?>"
+				aria-label="<?php p($l->t('Password')); ?>"
+				<?php p($_['user_autofocus'] ? '' : 'autofocus'); ?>
+				autocomplete="<?php p($_['login_form_autocomplete']); ?>" autocapitalize="none" autocorrect="off" required>
+			<label for="password" class="infield"><?php p($l->t('Password')); ?></label>
 		</p>
+
 		<div id="submit-wrapper">
-			<input type="submit" id="submit" class="login primary" title="" value="Login" disabled="disabled" />
+			<input type="submit" id="submit" class="login primary" title="" value="<?php p($l->t('Log in')); ?>" disabled="disabled" />
 			<div class="submit-icon icon-confirm-white"></div>
 		</div>
 
@@ -77,7 +83,6 @@
 		<?php } ?>
 
 		<div class="login-additional">
-			<!--
 			<?php if (!empty($_['canResetPassword'])) { ?>
 			<div class="lost-password-container">
 				<a id="lost-password" href="<?php p($_['resetPasswordLink']); ?>">
@@ -88,7 +93,6 @@
 				</a>
 			</div>
 			<?php } ?>
-		-->
 		</div>
 
 		<input type="hidden" name="timezone_offset" id="timezone_offset" />
