--- /var/ida/nextcloud/apps/files_sharing/js/dist/additionalScripts.js	2021-11-29 13:49:30.480612813 +0000
+++ /var/ida/nextcloud/apps/files_sharing/js/dist/additionalScripts.js.original	2021-11-29 13:49:02.080435453 +0000
@@ -454,7 +454,7 @@
             },
             _formatRemoteShare: function(e, t, r) {
                 var n = OCA.Sharing.Util._REMOTE_OWNER_REGEXP.exec(e);
-                if ( /*console.error(n),*/ !n || !n[7]) return '<span class="avatar" data-username="' + i()(e) + '" title="' + r + " " + i()(t) + '"></span><span class="hidden-visually">' + r + " " + i()(t) + "</span> ";
+                if (console.error(n), !n || !n[7]) return '<span class="avatar" data-username="' + i()(e) + '" title="' + r + " " + i()(t) + '"></span><span class="hidden-visually">' + r + " " + i()(t) + "</span> ";
                 var a = n[2],
                     s = n[4],
                     o = n[5],
