--- /var/ida/nextcloud/lib/private/Share/Constants.php	2021-11-29 13:44:49.897860561 +0000
+++ /var/ida/nextcloud/lib/private/Share/Constants.php.original	2021-11-03 10:50:55.013540263 +0000
@@ -83,7 +83,7 @@
 
 	public const RESPONSE_FORMAT = 'json'; // default resonse format for ocs calls
 
-	public const TOKEN_LENGTH = 12; // old (oc7) length is 32, keep token length in db at least that for compatibility
+	public const TOKEN_LENGTH = 15; // old (oc7) length is 32, keep token length in db at least that for compatibility
 
 	protected static $shareTypeUserAndGroups = -1;
 	protected static $shareTypeGroupUserUnique = 2;
