--- /var/ida/nextcloud-old/core/js/oc-dialogs.js	2021-10-27 10:25:31.029496617 +0000
+++ /var/ida/nextcloud-old/core/js/oc-dialogs.js.original	2021-10-27 10:25:31.029496617 +0000
@@ -83,25 +83,6 @@
 		);
 	},
 	/**
-	* displays alert dialog
-	* @param text content of dialog
-	* @param title dialog title
-	* @param callback which will be triggered when user presses YES or NO
-	*        (true or false would be passed to callback respectively)
-	* @param modal make the dialog modal
-	*/
-	alertHtml:function(text, title, callback, modal) {
-		return this.message(
-			text,
-			title,
-			'alert',
-			OCdialogs.OK_BUTTON,
-			callback,
-			modal,
-			true
-		);
-	},
-	/**
 	* displays confirmation dialog
 	* @param text content of dialog
 	* @param title dialog title
@@ -348,13 +329,8 @@
 
 				if (checkInput()) { 
                     var newname = $input.val();
-                    var targetDir = self.$filePicker.data('path');
-                    if (targetDir == null || targetDir == '' || targetDir == '/' || OCA.IDA.Util.testIfFrozen(targetDir)) {
-                        OC.Notification.show(t('ida', 'Files can be added only in the Staging area (root folder ending in +)'), {type: 'error'});
-                        return;
-                    }
 					self.filepicker.filesClient.createDirectory(self.$filePicker.data('path') + '/' + newname).always(function (status) {
-						self._fillFilePicker(self.$filePicker.data('path') + '/' + newname );
+						self._fillFilePicker(self.$filePicker.data('path') + newname );
 					});
 					OC.hideMenus();
 					self.$filePicker.ocdialog('unsetEnterCallback');
